rubbish/src/Mode.hs

```diff
@@ -0,0 +1,217 @@
+
+instance HasUserSecret RubbishContext where
+    userSecret = rcRealModeContext_L . userSecret
+
+instance HasShutdownContext RubbishContext where
+    shutdownContext = rcRealModeContext_L . shutdownContext
+
+instance HasNodeContext RubbishSscType RubbishContext where
+    nodeContext = rcRealModeContext_L . nodeContext
+
+instance HasSlottingVar RubbishContext where
+    slottingTimestamp = rcRealModeContext_L . slottingTimestamp
+    slottingVar = rcRealModeContext_L . slottingVar
+
+instance {-# OVERLAPPABLE #-}
+    HasLens tag (RealModeContext RubbishSscType) r =>
+    HasLens tag RubbishContext r
+  where
+    lensOf = rcRealModeContext_L . lensOf @tag
+
+instance HasLoggerName' RubbishContext where
+    loggerName = rcRealModeContext_L . loggerName
+
+instance HasSlogContext RubbishContext where
+    slogContext = rcRealModeContext_L . slogContext
+
+instance HasSlogGState RubbishContext where
+    slogGState = rcRealModeContext_L . slogGState
+
+instance HasJsonLogConfig RubbishContext where
+    jsonLogConfig = rcRealModeContext_L . jsonLogConfig
+
+instance (HasCoreConstants, MonadSlotsData ctx RubbishMode)
+      => MonadSlots ctx RubbishMode
+  where
+    getCurrentSlot = getCurrentSlotSum
+    getCurrentSlotBlocking = getCurrentSlotBlockingSum
+    getCurrentSlotInaccurate = getCurrentSlotInaccurateSum
+    currentTimeSlotting = currentTimeSlottingSum
+
+instance {-# OVERLAPPING #-} HasLoggerName RubbishMode where
+    getLoggerName = getLoggerNameDefault
+    modifyLoggerName = modifyLoggerNameDefault
+
+instance {-# OVERLAPPING #-} CanJsonLog RubbishMode where
+    jsonLog = jsonLogDefault
+
+instance MonadDBRead RubbishMode where
+    dbGet = dbGetDefault
+    dbIterSource = dbIterSourceDefault
+
+instance MonadDB RubbishMode where
+    dbPut = dbPutDefault
+    dbWriteBatch = dbWriteBatchDefault
+    dbDelete = dbDeleteDefault
+
+instance HasCoreConstants =>
+         MonadBlockDBGenericWrite (BlockHeader RubbishSscType) (Block RubbishSscType) Undo RubbishMode where
+    dbPutBlund = dbPutBlundDefault
+
+instance HasCoreConstants =>
+         MonadBlockDBGeneric (BlockHeader RubbishSscType) (Block RubbishSscType) Undo RubbishMode
+  where
+    dbGetBlock  = dbGetBlockDefault @RubbishSscType
+    dbGetUndo   = dbGetUndoDefault @RubbishSscType
+    dbGetHeader = dbGetHeaderDefault @RubbishSscType
+
+instance HasCoreConstants =>
+         MonadBlockDBGeneric (Some IsHeader) (SscBlock RubbishSscType) () RubbishMode
+  where
+    dbGetBlock  = dbGetBlockSscDefault @RubbishSscType
+    dbGetUndo   = dbGetUndoSscDefault @RubbishSscType
+    dbGetHeader = dbGetHeaderSscDefault @RubbishSscType
+
+instance MonadGState RubbishMode where
+    gsAdoptedBVData = gsAdoptedBVDataDefault
+
+instance HasCoreConstants => MonadBListener RubbishMode where
+    onApplyBlocks = onApplyBlocksStub
+    onRollbackBlocks = onRollbackBlocksStub
+
+-- FIXME: I preserved the old behavior, but it most likely should be
```

Comment:
It should! Please do it.

Reply:
Basically, I forgot to do it in CSL-803

Reply:
Well, I would do it if it was trivial, but it's not so trivial, because AFAIU there were two implementations so far: this one with genesis utxo and the one in wallet (which uses wallet-db).
I don't want this thing to know about wallet db and I don't want to do something something non-(absolutely)trivial that isn't really for this issue (it was for CSL-803 most likely).
So I would like to do it in another issue.

Reply:
Created https://issues.serokell.io/issue/CSL-1635